zbus_xmlgen/src/lib.rs

```diff
@@ -190,7 +194,8 @@ impl<'i> GenTrait<'i> {
         signals.sort_by(|a, b| a.name().partial_cmp(&b.name()).unwrap());
         for signal in &signals {
             let args = parse_signal_args(signal.args());
-            let name = to_identifier(&to_snakecase(signal.name().as_str()));
+            let name = collision_hander
+                .avoid_collision(to_identifier(&to_snakecase(signal.name().as_str())));
```

Comment:
I think the 2 steps should be combined into one. We don't really need a separate type for name collision, just one for creating identifiers.

Reply:
First I also tought this, but the `to_identifier` is also used in other places, eg. function parameters, and they don't collide with signals and properties. but

I could move the `to_identifier` function inside the `avoid_collision` but I thought this was it is more explict

Reply:
> First I also tought this

Ah, you're ahead of me. :)

> I could move the `to_identifier` function inside the `avoid_collision` but I thought this was it is more explict

Or you could have the new API use `to_identifier` itself so that caller doesn't need to care. I'd call the type `UniqueNameGenerator` and method `identifier_for_name`, which takes the name as arg, also taking care of `to_snakecase`. You can suggest a better name, just don't like collision in the name (sounds negative :laughing:).

> I was just wondering if there is a case where we also have collisions in arguments?

Doubtful but even if there was, it won't be global (i-e args of different methods can have the same name).

Reply:
I tried severl variants of combining operations. In the end I found it's best to keep the operations separate so I can combine these in the different cases:

- Functions and signals i need `to_snakecase`, `to_identifier` and `make_unique`
- For Properties I need  `to_snakecase`, `to_identifier`, and I need `make_unique` as a separate operation depending on read or write case.
- For arguments I only need `to_identifier`, optionally we could apply `make_unique` per function here but I don't see this needed.

So my solution now is to just rename the CollisionHandler to Idenfifier and have a make_unique method.

Reply:
> * Functions and signals i need `to_snakecase`, `to_identifier` and `make_unique`

As I wrote, it would be best to just abstract it all. The caller should just ask for a unique identifier for the name and that's it.

> * For Properties I need  `to_snakecase`, `to_identifier`, and I need `make_unique` as a separate operation depending on read or write case.

From what I can tell from your commit, it's basically doing the same thing. Of course, it would make sense to have a separate API for properties in `UniqueNameGenerator`, e.g `property_getter` and `property_setter` (or it could just be `property` with a boolean arg that tells the direction. Then `identifier_for_name` could just be `method`. 

> * For arguments I only need `to_identifier`, optionally we could apply `make_unique` per function here but I don't see this needed.

:+1:

Reply:
I am not sure what you are asking now.

As I see it the current implementation has several functions/methods (`to_snakecase`, `to_identifier` and `make_unique`) that can be combined ot form the final identifier.

The rules how these need to be combined for the different cases are applied locally where the element (`method`, `property`, `signal` and `arguments`) are processed.

The way I understand your proposal is to put the rules how to combine these into separate functions/methods and just call them? While I can do that I'm not convinced it is a good idea as it basically splits the rules into two locations.

Reply:
> The way I understand your proposal is to put the rules how to combine these into separate functions/methods and just call them, e.g. `to_method`,`to_signal`, `to_property` and `to_argument`

At the very least, it should have been clear, which names I proposed. :)

> While I can do that I'm not convinced it is a good idea as it basically splits the rules into two locations.

How so? They'll do pretty much the same thing as you're doing right now, just in one place/type. They will all use `to_snakecase` and `to_identifier`, for example.

Reply:
Let me know, if you'd want me to make the proposed changes in a separate commit on top.

Reply:
I did a test with these changes and the NetworkManager dbus xml, and it doesn't solve all the conflicts. I need to dig a bit deeper. I think I will add a test case using the NetworkManager interface description to make the issue more reproducible.

Reply:
@bachp this is the part that is still not resolved. You never answered my last question here. I know it's been many months since so I don't blame you for not remembering but let's try to conclude this only remaining issue,

Reply:
hello??